### ISSUE TITLE
Race condition on uploading recursively?

### ISSUE THREAD

**Issue by petemounce** [justmiles/go-markdown2confluence#12]
I was able to work around the issue I'm describing by pulling the source, changing lib/markdown.go to change Parallelism to 1, and re-uploading.

I think there's therefore a race-condition within the bit of code that creates the nested structure.

I have a folder structure like

```
docs
├── adrs
│   ├── arbitrary-file-persistence
│   │   ├── 20200114-arbitrary-file-persistence.md
│   │   └── resources
│   │       ├── arbitrary-file-persistence-cronjob.png
│   │       └── arbitrary-file-persistence-daemonset.png
│   ├── configmanagement
│   │   ├── 20200106-kubernetes-config.md
│   │   ├── 20200115-vault-secret-injection.md
│   │   └── 20200116-kubernetes-namespacing.md
│   ├── dependencies
│   │   ├── 20200108_loggingpackage.md
│   │   ├── 20200109_websocketpackage.md
│   │   └── 20200113_clipackage.md
│   ├── logging
│   │   ├── 20191213-initial-logging-design.md
│   │   ├── 20200110-es-indexes.md
│   │   └── resources
│   │       └── initial-platform-logging.png
│   ├── mesh
│   │   ├── 20200115-service-mesh.md
│   │   └── 20200117-istio.md
│   ├── metrics
│   │   ├── 20191218-inital-metrics.md
│   │   └── resources
│   │       └── 20191218-initialdesign.png
│   ├── platform
│   │   ├── 20191203-initial-platform-iteration.md
│   │   └── resources
│   │       ├── containerised-runtime.png
│   │       └── old-platform-runtime.png
│   ├── template.md
│   └── workflow
│       └── local
│           └── 20191219-local-kubernetes.md
├── debt
│   ├── 20200114-k8s-client-go-version.md
│   ├── 20200115-ca-bundle-injection.md
│   ├── 20200115-istio-system-components.md
│   └── 20200115-kubernetes-secrets.md
└── guides
    ├── building-docker-containers.md
    ├── cluster-setup-steps.md
    ├── istio.md
    ├── logging.md
    └── separate-runtime-and-worker-pods.md
```

When I run `markdown2confluence --endpoint https://my-account.atlassian.net/wiki --space SVP --username me --password secret --parent "Source controlled documentation" docs` I get

```
Source controlled documentation/adrs/arbitrary-file-persistence/20200114-arbitrary-file-persistence: https://improbableio.atlassian.net/wiki/x/z4A6KQ
Source controlled documentation/adrs/configmanagement/20200115-vault-secret-injection: https://improbableio.atlassian.net/wiki/x/xQA8KQ
Source controlled documentation/adrs/configmanagement/20200116-kubernetes-namespacing: https://improbableio.atlassian.net/wiki/x/HwMBKQ
Source controlled documentation/adrs/dependencies/20200108_loggingpackage: https://improbableio.atlassian.net/wiki/x/4oA6KQ
Source controlled documentation/adrs/configmanagement/20200106-kubernetes-config: https://improbableio.atlassian.net/wiki/x/1gA8KQ
Source controlled documentation/adrs/dependencies/20200109_websocketpackage: https://improbableio.atlassian.net/wiki/x/qoA-KQ
Source controlled documentation/adrs/dependencies/20200113_clipackage: https://improbableio.atlassian.net/wiki/x/LQP_K
Source controlled documentation/adrs/mesh/20200117-istio: https://improbableio.atlassian.net/wiki/x/s4BAKQ
Source controlled documentation/adrs/mesh/20200115-service-mesh: https://improbableio.atlassian.net/wiki/x/6gA8KQ
Source controlled documentation/adrs/logging/20200110-es-indexes: https://improbableio.atlassian.net/wiki/x/fQBBKQ
Source controlled documentation/adrs/metrics/20191218-inital-metrics: https://improbableio.atlassian.net/wiki/x/MgMBKQ
Source controlled documentation/adrs/logging/20191213-initial-logging-design: https://improbableio.atlassian.net/wiki/x/3AA9KQ
Source controlled documentation/adrs/template: https://improbableio.atlassian.net/wiki/x/RYE8KQ
ERRO[0006] com.atlassian.confluence.api.service.exceptions.BadRequestException: A page with this title already exists: A page already exists with the title debt in the space with key SVP
Source controlled documentation/debt/20200115-istio-system-components:
Source controlled documentation/adrs/platform/20191203-initial-platform-iteration: https://improbableio.atlassian.net/wiki/x/5wfXK
Source controlled documentation/debt/20200114-k8s-client-go-version: https://improbableio.atlassian.net/wiki/x/SgMBKQ
Source controlled documentation/debt/20200115-kubernetes-secrets: https://improbableio.atlassian.net/wiki/x/qwA6KQ
Source controlled documentation/debt/20200115-ca-bundle-injection: https://improbableio.atlassian.net/wiki/x/VQMBKQ
Source controlled documentation/adrs/workflow/local/20191219-local-kubernetes: https://improbableio.atlassian.net/wiki/x/SwP_K
ERRO[0007] com.atlassian.confluence.api.service.exceptions.BadRequestException: A page with this title already exists: A page already exists with the title guides in the space with key SVP
Source controlled documentation/guides/cluster-setup-steps:
Source controlled documentation/guides/logging: https://improbableio.atlassian.net/wiki/x/XYL_K
Source controlled documentation/guides/building-docker-containers: https://improbableio.atlassian.net/wiki/x/y4A-KQ
Source controlled documentation/guides/istio: https://improbableio.atlassian.net/wiki/x/agMBKQ
Source controlled documentation/guides/separate-runtime-and-worker-pods: https://improbableio.atlassian.net/wiki/x/cAA_KQ

Unable to upload markdown file, docs/debt/20200115-istio-system-components.md:
	Error creating parent page docs/debt/20200115-istio-system-components.md for debt: com.atlassian.confluence.api.service.exceptions.BadRequestException: A page with this title already exists: A page already exists with the title debt in the space with key SVP

Unable to upload markdown file, docs/guides/cluster-setup-steps.md:
	Error creating parent page docs/guides/cluster-setup-steps.md for guides: com.atlassian.confluence.api.service.exceptions.BadRequestException: A page with this title already exists: A page already exists with the title guides in the space with key SVP
```

and a space structure that looks like 
![Screenshot 2020-01-27 at 15 11 06](https://user-images.githubusercontent.com/69720/73186222-453e8580-4117-11ea-953c-9c1794ef347e.png)

**Comment by justmiles** [comment]
Yep, it's trying to create the parent pages at the same time. I'll see if I can find a way to de-dupe the parent pages that need to be created.

**Comment by justmiles** [comment]
This is fixed in [v3.1.1](https://github.com/justmiles/go-markdown2confluence/releases/tag/v3.1.1)

**Comment by petemounce** [comment]
Thank you, that works beautifully!
